ui/components/src/model/units/time.ts

```diff
@@ -74,16 +75,21 @@ export enum TimeIntlDuration {
 }
 
 export function formatTime(value: number, { kind, decimal_places }: TimeUnitOptions): string {
-  const decimalPlaces = decimal_places ?? DEFAULT_DECIMAL_PLACES;
-
-  const timeUnit: string = TimeIntlDuration[kind];
   const isMonthOrYear = kind === 'Months' || kind === 'Years';
 
-  const formatter = new Intl.NumberFormat('en-US', {
+  const formatterOptions: Intl.NumberFormatOptions = {
     style: 'unit',
-    unit: timeUnit,
+    unit: PersesTimeToIntlTime[kind],
     unitDisplay: isMonthOrYear ? 'long' : 'narrow',
-    maximumFractionDigits: decimalPlaces,
-  });
+  };
+
+  const hasDecimalPlaces = decimal_places !== undefined && decimal_places !== null;
+  if (hasDecimalPlaces) {
+    formatterOptions.maximumFractionDigits = limitDecimalPlaces(decimal_places);
+  } else {
+    formatterOptions.maximumSignificantDigits = MAX_SIGNIFICANT_DIGITS;
+  }
```

Comment:
It looks like you do this little bit of code (or something very similar) ~3 times in here. Is this a case where it's slightly different bits of code that I pattern matched too quickly or are these identical. If the latter, maybe worth pulling out into a little formatting util, so if we adjust these again it can happen in one place.

Reply:
Oo yes I will give that try